### ISSUE TITLE
TypeError: unsupported operand type(s) for -: 'NoneType' and 'int'

### ISSUE THREAD

**Issue by rozetko** [hastic/hastic-server#430]
```
learning started...

query time range: from Fri Feb 15 2019 00:28:35 GMT+0530 (IST) to Thu Feb 21 2019 08:10:34 GMT+0530 (IST)

run task, id:nXfUyBpjgEqlZ83T

2019-02-21 08:10:35,068 [Analytics] [INFO ]  Got LEARN task with id gNFJcQkmEx1bpCEv, analyticUnitId nXfUyBpjgEqlZ83T

 

2019-02-21 08:10:35,262 [Analytics] [ERROR]  handle_analytic_task Exception: 'Traceback (most recent call last):

  File "/home/rozetko/hastic-server/analytics/analytics/analytic_unit_manager.py", line 94, in handle_analytic_task

  File "/home/rozetko/hastic-server/analytics/analytics/analytic_unit_manager.py", line 82, in __handle_analytic_task

  File "/home/rozetko/hastic-server/analytics/analytics/analytic_unit_worker.py", line 28, in do_train

  File "/usr/lib64/python3.6/concurrent/futures/thread.py", line 56, in run

  File "/home/rozetko/hastic-server/analytics/analytics/detectors/pattern_detector.py", line 46, in train

  File "/home/rozetko/hastic-server/analytics/analytics/models/model.py", line 69, in fit

  File "/home/rozetko/hastic-server/analytics/analytics/models/model.py", line 19, in __init

TypeError: unsupported operand type(s) for -: 'NoneType' and 'int'
```

**Comment by amper43** [comment]
Fixed in #427

**Comment by amper43** [comment]
Fixed in #427
